### DISCUSSION TITLE
Tauri how to render different url adress ?

### DISCUSSION THREAD

**Discussion by erikpa1** [tauri-apps/tauri#9146] [Q&A]
Hi, I know it was in documentation, but doc changed a little bit and now I don't see it. Thing is, I want to migrate my python / flask app to tauri. (I run it now via py2exe + web browser). And I want to create python subprocess and link tauri to it (via localhost:5000).

I can't  migrate it at once,.. so I have to create small parts, which will be slowly exchanged until python die totally.

How can I make tauri to go for app on for example localhost:5000 ?? I know there is in tauri.config URL, but when app is built, it takes static files from ../build anyway.

**Comment by FabianLars** [comment]
If you don't need any static assets and serve _everything_ over the server you can change devPath and distUrl (v1) / devUrl and frontendDist (v2) to that localhost url.
If you want to do that but you have a dynamic port you can do it like this: https://github.com/tauri-apps/tauri-plugin-localhost?tab=readme-ov-file#usage (you don't need the plugin, it's just the best example)

You can also specify it per window with the window specific url configs https://tauri.app/v1/api/config/#windowconfig.url / https://docs.rs/tauri/latest/tauri/window/struct.WindowBuilder.html#method.new / https://tauri.app/v1/api/js/window#constructor-5

To navigate at runtime you'll have to use window.location in js in v1 and https://docs.rs/tauri/2.0.0-beta.10/tauri/webview/struct.WebviewWindow.html#method.navigate (or window.location) in v2.

**Reply by erikpa1** [reply]
@FabianLars YES, thanks, those I was searching for. But the   window.config URL wasn't working after `pnpm tauri build` , there was testing index.html, I had in build. There wasn't my localhost:5000

**Reply by erikpa1** [reply]
@FabianLars  Sorry, I tried it home and "url" is working, maybe some cache , or something went bad on my work PC,... thanks again, have a nice day :)

**Comment by doorlay** [comment]
@FabianLars is there a method for getting the current URL after rendering an external URL? My use case is that I'd like to render an external URL, have the user take action within the web page, and then check what the current URL is post-action.

**Reply by FabianLars** [reply]
On the rust side you can use https://docs.rs/tauri/latest/tauri/window/struct.Window.html#method.url (v1) / https://docs.rs/tauri/2.0.0-beta/tauri/webview/struct.WebviewWindow.html#method.url (v2) at any time to receive the current url.

To know when to trigger it may be tricky though and depends on what kind of action we're talking about and if you can hook into that.
